File: python/matching-brackets/matching_brackets.py
BRACKETS = {'[': ']', '{': '}', '(': ')'}


def is_paired(input_string):
    """
    Valid bracket sentence
    """
    stack = []
    open = BRACKETS.keys()
    close = BRACKETS.values()
    for s in input_string:
        if s in open:
            stack.append(s)

        if s in close:
            if not stack or BRACKETS[stack.pop()] != s:
                return False

    return not stack

File: javascript/matching-brackets/matching-brackets.js
export const isPaired = (input) => {
  const MAP_BRACKETS = {'}': '{', ']': '[', ')': '('}; 
  const OPENING = "{[(";
  const stack = [];

  for (const s of input){
    if (OPENING.includes(s)) stack.push(s); 
    else if (!!MAP_BRACKETS[s] && stack && stack.pop() !== MAP_BRACKETS[s])
        return false;
  }
  return stack.length === 0;
};
